File: Coding in Python/26th January 2026/Minimum Absolute Difference.py
#Minimum Absolute Difference
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        res = []
        arr.sort()
        diff = arr[1] - arr[0]
        for i in range(1, len(arr)):
            if arr[i] - arr[i - 1] < diff:
                diff = arr[i] - arr[i - 1]
        for i in range(1, len(arr)):
            if arr[i] - arr[i - 1] == diff:
                temp = []
                temp.append(arr[i - 1])
                temp.append(arr[i])
                res.append(temp)
        return res

File: coding/26th January 2026/Minimum Absolute Difference.java
//Minimum Absolute Difference
class Solution 
{
    public List<List<Integer>> minimumAbsDifference(int[] arr) 
    {
        Arrays.sort(arr);
        List<List<Integer>> list = new ArrayList<>();
        int diff = arr[1] - arr[0];
        for(int i = 1; i < arr.length; i++)
        {
            if(arr[i] - arr[i - 1] <= diff)
                diff = arr[i] - arr[i - 1];
        } 
        for(int i = 1; i < arr.length; i++)
        {
            if(arr[i] - arr[i - 1] == diff)
            {
                ArrayList<Integer> temp = new ArrayList<>();
                temp.add(arr[i - 1]);
                temp.add(arr[i]);
                list.add(temp);
            }
        }
        return list;
    }
}
